Tasks/UseRubyV1/installer.ts

```diff
@@ -0,0 +1,54 @@
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+
+import * as task from "azure-pipelines-task-lib/task";
+import * as tool from "azure-pipelines-tool-lib/tool"
+
+enum Platform {
+    Windows,
+    MacOS,
+    Linux
+}
+
+function getPlatform(): Platform {
+    switch (process.platform) {
+        case "win32": return Platform.Windows;
+        case "darwin": return Platform.MacOS;
+        case "linux": return Platform.Linux;
+        default: throw Error(task.loc("PlatformNotRecognized"));
+    }
+}
+
+interface TaskParameters {
+    readonly version: string;
+    readonly architecture: string;
+}
+
+export async function installRubyVersion(parameters: TaskParameters): Promise<void> {
+    const toolName: string = "Ruby";
+    const installDir: string | null = tool.findLocalTool(toolName, parameters.version, parameters.architecture);
+    if (!installDir) {
```

Comment:
This doesn't make sense to me - why are we throwing if the tool isn't found locally? Shouldn't we download if its not found?

Reply:
And cache it once we download it.

Reply:
I guess this is what the existing task is doing, it feels like we should be able to do better than that though.

Reply:
Double checking with @vtbassmatt 

I think the idea was to preserve behavior for now and extend later

Reply:
Ok - if we do decide to download it looks like the dist is pretty well structured - https://cache.ruby-lang.org/pub/ruby/

Can probably basically just do the same thing we do in the UseNode task

Reply:
As far as I can tell those are not prebuilt binaries and I doubt we can afford the time to build ruby on demand. Ill dig in a little more, but I believe lack of simple multiplat binary distributions is part of why the existing task doesnt have this functionality

Reply:
Oh yeah, you're right 😕

I still think its weird that we download other tools when they're not present, but not Ruby. It looks like the plan is to do the same with Python though.

With that said, I don't see a good way around this so I guess I'm ok with it.

Reply:
Now that this is "done" Ill check with Matt on timelines. I think the on-demand install is a nice-to-have so we may choose to go for it, and see if its viable to get done within the timeline. I agree  with you that being consistent would be ideal, and as far as I'm aware there's no licensing issues or anything blocking this. 

I would probably say lets go ahead and get this merged if it looks correct, even if we end up immediately revisiting and extending it

Reply:
Agreed

Reply:
In an ideal world, we'd always download any version the customer asked for. In practice that's not always been easy/possible, so we've fallen back to just the tools cache. @dakale is right that we need to at least match parity, making improvements where possible but not blocking on them.